From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Thu, 15 Sep 2022 12:51:44 +0200
Subject: [PATCH] imx8mp-irma6r2.dts: Serializer config

Use correct PWM timings and set tc358746 clock to non-continuous mode
Set hs-settle accordingly to the 216Mhz CSI clock frequency

Signed-off-by: Erik Schumacher <erik.schumacher@iris-sensing.com>
Signed-off-by: Ian Dannapel <ian.dannapel@iris-sensing.com>
---
 arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts b/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts
index 40204717416f..e5b329052395 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts
@@ -70,9 +70,9 @@
 	clk_serializer: serializer_clock {
 		compatible = "pwm-clock";
 		#clock-cells = <0>;
-		clock-frequency = <24000000>;
+		clock-frequency = <23800000>;
 		clock-output-names = "serializer_clk";
-		pwms = <&pwm2 0 42>; /* 42ns - 24 MHz */
+		pwms = <&pwm2 0 42>; /* 42ns - 23.8 MHz */
 	};
 
 	gpio-keys {
@@ -245,6 +245,7 @@
 			reg = <1>;
 			tc358748_mipi_out: endpoint {
 				data-lanes = <1 2>;
+				clock-noncontinuous;
 				link-frequencies = /bits/ 64 <216000000>;
 				remote-endpoint = <&mipi1_csi_in>;
 			};
@@ -268,7 +269,7 @@
 			data-lanes = <2>;
 			clock-lanes = <1>;
 			remote-endpoint = <&tc358748_mipi_out>;
-			csis-hs-settle = <3>;
+			csis-hs-settle = <4>;
 			csis-clk-settle = <0>;
 			csis-wclk;
 			link-frequencies = /bits/ 64 <216000000>;
-- 
2.25.1

